homeassistant/components/google_photos/media_source.py

```diff
@@ -47,54 +48,70 @@
 RECENT_PHOTOS_TITLE = "Recent Photos"
 
 
+class PhotosIdentifierType(StrEnum):
+    """Type for a PhotosIdentifier."""
+
+    PHOTO = "p"
+    ALBUM = "a"
+
+    @classmethod
+    def of(cls, name: str) -> "PhotosIdentifierType":
+        """Parse a PhotosIdentifierType by string value."""
+        for enum in PhotosIdentifierType:
+            if enum.value == name:
+                return enum
+        raise ValueError(f"Invalid PhotosIdentifierType: {name}")
+
+
 @dataclass
 class PhotosIdentifier:
     """Google Photos item identifier in a media source URL."""
 
     config_entry_id: str
     """Identifies the account for the media item."""
 
-    album_media_id: str | None = None
-    """Identifies the album contents to show.
-
-    Not present at the same time as `photo_media_id`.
-    """
-
-    photo_media_id: str | None = None
-    """Identifies an indiviidual photo or video.
+    id_type: PhotosIdentifierType | None = None
+    """Type of identifier"""
 
-    Not present at the same time as `album_media_id`.
-    """
+    media_id: str | None = None
+    """Identifies the album or photo contents to show."""
 
     def as_string(self) -> str:
         """Serialize the identiifer as a string.
 
-        This is the opposite if parse_identifier().
+        This is the opposite if of().
         """
-        if self.photo_media_id is None:
-            if self.album_media_id is None:
-                return self.config_entry_id
-            return f"{self.config_entry_id}/{PHOTO_SOURCE_IDENTIFIER_ALBUM}/{self.album_media_id}"
-        return f"{self.config_entry_id}/{PHOTO_SOURCE_IDENTIFIER_PHOTO}/{self.photo_media_id}"
+        if self.id_type is None:
+            return self.config_entry_id
+        return f"{self.config_entry_id}/{self.id_type}/{self.media_id}"
+
+    @staticmethod
```

Comment:
All these methods should be `@classmethod`.

Reply:
Fixing in #124977